src/Microsoft.AspNetCore.ResponseCompression/CompressionProviderFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.ResponseCompression
+{
+    /// <summary>
+    /// This is a placeholder for the CompressionProviderCollection that allows creating the given type via
+    /// an <see cref="IServiceProvider" />.
+    /// </summary>
+    internal class CompressionProviderFactory : ICompressionProvider
+    {
+        internal CompressionProviderFactory(Type providerType)
```

Comment:
It seems to me that logically the ctor, properties, and methods on this type should all be public, no? I realize the type is internal, but as far as member visibility goes the pattern we typically follow is, "what visibility would this have *if* the type was public"? And I think in this case they'd be public.